tracer.go

```diff
@@ -112,7 +112,7 @@ func (r *tracerS) StartSpanWithOptions(operationName string, opts ot.StartSpanOp
 	return &spanS{
 		context:     sc,
 		tracer:      r,
-		Service:     sensor.options.Service,
+		Service:     sensor.serviceName,
```

Comment:
This is the actual fix: Add the service name as an annotation, no matter whether it came from the env var or the in-code configuration.